src/app/[locale]/components/Tasks.txt

```diff
@@ -0,0 +1,152 @@
+"use client";
+import Link from "next/link";
+import { MdDelete, MdEdit } from "react-icons/md";
+import { useEffect, useMemo, useState } from "react";
+import { useTranslations } from "next-intl";
+import { selectAuth } from "../../../redux/slices/authSlice";
+import { useDispatch, useSelector } from "react-redux";
+import { AppDispatch } from "../../../redux/store";
+import {
+  deleteTask,
+  getAllTasks,
+  selectTask,
+  setTasks,
+} from "../../../redux/slices/taskSlice";
+import { useRouter } from "next/navigation";
+import { TaskData } from "../types/typescript";
+
+export default function Tasks({
+  locale,
+  initialTasks,
+}: {
+  locale: string;
+  initialTasks: TaskData[];
+}) {
+  const dispatch = useDispatch<AppDispatch>();
+  const { isLoggedIn } = useSelector(selectAuth);
+  const { tasks, deleteFlag } = useSelector(selectTask);
+  const [authorisedUser, setAuthorisedUser] = useState<boolean>();
+  const [admin, setAdmin] = useState<boolean>();
+  const t = useTranslations("translations");
+  const router = useRouter();
+
+  const filteredTasks = useMemo(() => {
+    if (!isLoggedIn) return [];
+    return tasks.filter((task) => {
+      if (
+        isLoggedIn.roles.includes("admin") ||
+        isLoggedIn.roles.includes("manager")
+      ) {
+        return true;
+      }
+      return (
+        task.assignedTo?.toString() === isLoggedIn._id.toString() ||
+        task.group?.toString() === isLoggedIn.group?.toString()
+      );
+    });
+  }, [tasks, isLoggedIn]);
+
+  useEffect(() => {
+    if (!isLoggedIn) {
+      router.push(`/${locale}`);
+    }
+    if (
+      isLoggedIn?.roles.includes("admin") ||
+      isLoggedIn?.roles.includes("manager")
+    ) {
+      setAuthorisedUser(true);
+    }
+    if (isLoggedIn?.roles.includes("admin")) {
+      setAdmin(true);
+    }
+    if (tasks.length > 0) {
+      dispatch(getAllTasks());
+    } else {
+      dispatch(setTasks(initialTasks));
+    }
+  }, []);
```

Comment:
shouldnt there be dependency arrays in here  so it only re-runs the relevent dependencies?